src/stores/viewStore/createViewFilterSlice.ts

```diff
@@ -1,53 +1,52 @@
 import {StateCreator} from "zustand";
 
-import {LogLevelFilter} from "../../typings/logs";
-import {UI_STATE} from "../../typings/states";
-import {CURSOR_CODE} from "../../typings/worker";
+import {updateWindowUrlHashParams} from "../../utils/url";
 import useLogFileManagerStore from "../logFileManagerProxyStore";
 import {handleErrorWithNotification} from "../notificationStore";
-import useQueryStore from "../queryStore";
-import useUiStore from "../uiStore";
 import {
     ViewFilterSlice,
+    ViewFilterValues,
     ViewState,
 } from "./types";
 
 
+const VIEW_FILTER_DEFAULT: ViewFilterValues = {
+    logLevelFilter: null,
+    kqlFilter: "",
+    isFilterApplied: false,
+};
+
 /**
  * Creates a slice for view utility functions.
  *
- * @param _
+ * @param set
  * @param get
  * @return
  */
 const createViewFilterSlice: StateCreator<
     ViewState, [], [], ViewFilterSlice
-> = (_, get) => ({
-    filterLogs: (filter: LogLevelFilter) => {
-        const {setUiState} = useUiStore.getState();
-        setUiState(UI_STATE.FAST_LOADING);
+> = (set, get) => ({
+    ...VIEW_FILTER_DEFAULT,
+    filterLogs: () => {
+        const {logLevelFilter, kqlFilter} = get();
         (async () => {
             const {logFileManagerProxy} = useLogFileManagerStore.getState();
```

Comment:
```suggestion
        (async () => {
            const {logLevelFilter, kqlFilter} = get();
            const {logFileManagerProxy} = useLogFileManagerStore.getState();
```
I think it is best to put the get() statement inside the async() so that it will be least affected.